src/utils/navigatorData.js

```diff
@@ -179,16 +179,21 @@ export function getSiblings(uid, childrenMap, children) {
 }
 
 function extractRootNode(data) {
+  // note: this "root" path won't always necessarily come at the beginning of
+  // the URL in situations where the renderer is being hosted at some path
+  // prefix
+  const rootPathPattern = /^.*(\/documentation\/[^/]+).*$/;
+  const rootPath = window.location.href.match(rootPathPattern)?.[1] ?? '';
   // most of the time, it is expected that `data` always has a single item
   // that represents the top-level root node of the navigation tree
   //
   // there may be rare, unexpected scenarios where multiple top-level root
-  // nodes are provide for some reason—if that happens, we would prefer the
-  // first one categorized as a "module"
+  // nodes are provide for some reason—if that happens, we would prefer the one
+  // with a path that most closely resembles the current URL path
   //
   // otherwise, the first provided node will be used
   return data.length === 1 ? data[0] : (data.find(node => (
-    node.type === TopicTypes.module
```

Comment:
Is there any risk of a regression from removing the “always prefer the node whose `type` is `module`” behaviour, or do we now believe that it was a special case of the more generally correct behaviour (“always prefer the node whose `path` best matches the current page’s URL”) all along?

Reply:
The module check was done only recently (#916), and it has turned out to not be sufficient, resulting in the need for this PR. In the past, a similar check was done before some major refactoring was done to the sidebar components, so I'm not worried about regressions from removing the module check specifically.